app/domain/repos/conjur_ca.rb

```diff
@@ -14,12 +14,26 @@ def self.create(resource_id)
     #
     def self.ca(resource_id)
       ca_info = ::Conjur::CaInfo.new(resource_id)
-      stored_cert = Resource[ca_info.cert_id].last_secret.value
-      stored_key = Resource[ca_info.key_id].last_secret.value
+      stored_cert = fetch_ca_resource(ca_info.cert_id)
+      stored_key = fetch_ca_resource(ca_info.key_id)
```

Comment:
this will make 2 calls to the DB instead of 1 so it's not the ideal solution. I would retrieve them together and then take each var from the retrieved list. you can see how we did it in `FetchAuthenticatorSecrets`.